### ISSUE TITLE
Glide integration

### ISSUE THREAD

**Issue by stefan-niedermann** [nextcloud/Android-SingleSignOn#172]
For Deck and Notes we added a `GlideModule` to smoothly allow loading avatars and protected stuff via the SSO API.

It would be nice if the integration would be bundled with SSO since i know, that News app also uses Glide to fetch the avatar (i wrote this part :laughing: ).

Does this make sense for you or don't you want to be that specific in this repository?

Sample implementation in Notes (does not depend on any code outside of this package): https://github.com/stefan-niedermann/nextcloud-notes/tree/666-glide-sso/app/src/main/java/it/niedermann/owncloud/notes/util/glide

**Comment by tobiasKaminsky** [comment]
I would rather put it in readme.md than in library, as there might be apps that do not make use of it.
(or is this then automatically stripped off?)

**Comment by stefan-niedermann** [comment]
Hmm...

Glide handles this with an `integration`-subfolder for some HTTP-libraries: https://github.com/bumptech/glide/tree/master/integration

We could ask them to add this, but i doubt they would do so.

Or we could add an integration folder to this project, but i am not sure if this is stripped of, if glide is not used.

Another approach would be a completely separate project, but this would also have to be maintained etc...

But you are right: A section in the README would be the quickest solution (just link and instruct to copy those few files)...

**Comment by David-Development** [comment]
> Or we could add an integration folder to this project, but i am not sure if this is stripped of, if glide is not used.

I think this would also be my preferred solution. We'd have to find out first if it is stripped if it is not used.

**Comment by stefan-niedermann** [comment]
https://github.com/bumptech/glide/wiki/Integration-Libraries#why-isnt-xxx-library-implemented

**Comment by stefan-niedermann** [comment]
Let's see what the Glide-team says. In the meantime i published an own library which everyone who wants to can use. If it is ready, i will add a hint in the SSO documentation via PR.

-> i agree that this might be out of scope for the sso lib itself -> closing this issue

**Comment by stefan-niedermann** [comment]
So, the Glide team declined a direct integration in their repository.

In the meantime 3 Apps (Deck, Notes and News) are using Glide in combination with SingleSignOn. While Deck & Notes have up to date duplicate implemenations, News didn't receive the latest bugfixes yet.

The Glide integration has been moved to a separate Android library module in Deck and Notes. I tried to fork the SSO lib and add the Glide integration, but i was not able to access this module via Jitpack.

Anyone interested in discussing this topic again, so we do not have duplicate code for each project? I would also be open for a separate project (though this should not be hosted at my private GitHub account).
